module_utils/certificate/providers/certmonger.py

```diff
@@ -0,0 +1,142 @@
+import os
+
+from io import StringIO
+
+# pylint: disable=no-name-in-module
+from ansible.module_utils.six.moves.configparser import ConfigParser
+
+from .base import CertificateRequestBaseProvider
+
+
+class CertificateRequestCertmongerProvider(CertificateRequestBaseProvider):
+    """Certmonger provider for certificate Linux System Role."""
+
+    def __init__(self, *args, **kwargs):
+        super(CertificateRequestCertmongerProvider, self).__init__(*args, **kwargs)
+
+        self.requests_dir = "/var/lib/certmonger/requests"  # TODO: Allow to override?
+        self._request_file_path = self._find_certmonger_request_file()
+        self._certmonger_metadata = self._read_certmonger_request_file()
+
+    def _find_certmonger_request_file(self):
+        """Search for certificate metadata in certmonger data directory.
+
+        Looks for the certificate path, file by file inside certmonger data
+        directory.
+
+        Only open files that are 50k or smaller to avoid getting stuck for
+        a long time.
+
+        If the file is found it's immediately. If not found returns None.
+        """
+        ls_results = os.listdir(self.requests_dir)
+        line_start = "cert_storage_location="
+        for fname in ls_results:
+            file_path = os.path.join(self.requests_dir, fname)
+            fsize = os.stat(file_path).st_size
+            # do not look into file if it's larger than ~50k
+            #   certmonger request files usually avg around 8k
+            if fsize > (50000):
```

Comment:
This seems rather arbitrary - also, why `(50000)` instead of `50000`?

Reply:
It is indeed a somehow arbitrary. The point is that if we someone places a big file in this folder the role would open this file looking for information. It could be 10k but in case the user requests a larger certificate the file might be larger then 10k. 50k just gives enough margins  for that.

The parenthesis are there for no reason. I'll remove it.

Reply:
Is the problem that `requests_dir` may usually contain lots of different kinds of files, are not discriminated by a file name extension (e.g. .conf), or can have the type determined by looking at the first few 100 or so bytes of the file?  Or is the issue that a malicious user may try a DoS attack by adding a 100GB file to this directory?  I'm just trying to understand the use case here.

Reply:
Originally the directory should only contain text files and the file names do not have extension. The file usually starts with `id=...` but that's not always true. 
I've just checked with @rcritten and he said it should be safe to trust that everything in that directory is a certmonger request file.

Reply:
Removed in 0754286301d679dcb2559f2ca6c4f8e9f805776f